src/linked_hash_map.rs

```diff
@@ -297,6 +297,35 @@ impl<K: Hash<HmHasher> + Eq, V> LinkedHashMap<K, V> {
         }
     }
 
+    /// A double-ended iterator visiting all key-value pairs in order of insertion.
+    /// Iterator element type is `(&'a K, &'a mut V)`
+    /// # Examples
+    /// ```rust
+    /// use collect::LinkedHashMap;
+    ///
+    /// let mut map = LinkedHashMap::new();
+    /// map.insert("a", 10);
+    /// map.insert("c", 30);
+    /// map.insert("b", 20);
+    ///
+    /// {
+    ///     let mut iter = map.iter_mut();
+    ///     let mut entry = iter.next().unwrap();
+    ///     assert_eq!(&"a", entry.0);
+    ///     *entry.1 = 17;
+    /// }
+    ///
+    /// assert_eq!(&17, map.get(&"a").unwrap());
+    /// ```
+    pub fn iter_mut<'a>(&'a mut self) -> IterMut<'a, K, V> {
```

Comment:
I just meant that the iter_mut method doesn't need to be explicitly parameterized with a a lifetime. The struct and impls need it.